### DISCUSSION TITLE
Custom Fields on Interface

### DISCUSSION THREAD

**Discussion by MrPaulAR** [netbox-community/netbox#5982] [General]
I'm looking for a way to track supported speeds on some of my interfaces.  I have been going by the media type but unfortunately that results in me netbox assumptions that a 1000BaseT will be either 1000, 100/1000, or 10/1000/1000.  I also looked into adding a custom field to the interface but there wasn't an option for dcim > interfaces.  

I also looked at circuits > circuit but on the WebUI device > interfaces view it doesn't show up for the end users and I don't see an attribute that I can add to display it.

A similar issue we are working through is  we have some shared ports where one logical interface can be configured via software to use RJ45 or SFP interface type and we plan on having both cabled.  Due to this I've created Gi0_RJ45 and Gi0_SFP and I'll programmatically drop the extension.  I'll take advantage of the "enabled" flag for this.  Custom fields may be useful here too.

**Comment by DanSheps** [comment]
Config Contexts might do the trick, also custom fields are coming to interfaces in the next semi-major release (2.11)

**Reply by MrPaulAR** [reply]
Nice.  I'll hold off then.  That next release in the next month or so?

**Reply by DanSheps** [reply]
We are currently waiting for a few upstream projects to finish updates on their end and then will most likely deploy with that.
